src/stream/src/executor/merge.rs

```diff
@@ -172,6 +178,17 @@ impl MergeExecutor {
                         select_all.remove_upstreams(
                             &update.removed_upstream_actor_id.iter().copied().collect(),
                         );
+
+                        let col_idxes = select_all
+                            .untransferred_watermarks
+                            .keys()
+                            .cloned()
+                            .collect_vec();
+                        for col_idx in col_idxes {
+                            if let Some(watermark) = select_all.check_heap(col_idx) {
+                                yield Message::Watermark(watermark);
+                            }
+                        }
```

Comment:
Is this necessary? As there must be watermark messages after the scaling, we can defer this after the barrier.

Reply:
> Is this necessary? As there must be watermark messages after the scaling, we can defer this after the barrier.

The earlier the better for watermark messages

Reply:
Actually, we prefer to yield fewer messages during the scaling to finish the process as soon as possible. Both methods LGTM, not a big deal.